lib/src/kbasesearchengine/system/TypeStorage.java

```diff
@@ -2,16 +2,37 @@
 
 import java.util.List;
 
-import kbasesearchengine.events.exceptions.IndexingException;
-
+/** Stores type documents that describe how to transform an object from a a data source into
+ * one or more documents that can be indexed by search and mappings from data source type versions
+ * to search transformation document versions.
+ * @author gaprice@lbl.gov
+ *
+ */
 public interface TypeStorage {
     
+    /** Returns a transformation document given a search transformation type and version.
+     * @param type the search transformation type.
+     * @return the transformation document.
+     * @throws NoSuchTypeException if no such type exists.
+     */
     ObjectTypeParsingRules getObjectType(SearchObjectType type)
```

Comment:
getObjectTypeParsingRules instead of getObjectType?

Reply:
changed